pkg/networkservice/core/next/tail_client.go

```diff
@@ -25,10 +25,15 @@ import (
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 )
 
-// tailServer is a simple implementation of networkservice.NetworkServiceServer that is called at the end of a chain
-// to insure that we never call a method on a nil object
 type tailClient struct{}
 
+// TailClient is a simple implementation of networkservice.NetworkServiceClient that is:
+// - called at the end of a chain to insure that we never call a method on a nil object
+// - used to break an existing chain
+func TailClient() networkservice.NetworkServiceClient {
```

Comment:
Why do we need to make this public?

Reply:
Actually it is not used here, but same as `next.TailServer()` it can be used to get distinct chain elements out of the chain context:
```
... -> wrapper (element) -> ...
```
example wrapper it a _reset on double request_ wrapper:
``` go
if wr.isDoubleRequest {
    _, _ = chain.NewClient(element, next.TailClient()).Close(ctx, request.GetConnection())
} else {
    wr.isDoubleRequest = true
}
return element.Request(ctx, request)
```
it only closes `element` but not all subsequent chain.

Reply:
Can we rollback tails element back and create other ones for `timeout` goals?

Reply:
Replaced with `chainbreak`.